runtime/src/bank.rs

```diff
@@ -1210,16 +1211,66 @@ impl Bank {
         }
     }
 
+    fn distribute_rent_to_validators(
+        &self,
+        vote_account_hashmap: &HashMap<Pubkey, (u64, Account)>,
+        rent_to_be_distributed: u64,
+    ) {
+        let mut node_stake_hashmap: HashMap<Pubkey, u64> = HashMap::new();
+        let mut total_staked = 0;
+
+        vote_account_hashmap
+            .iter()
+            .map(|(_vote_pubkey, (staked, account))| {
+                total_staked += *staked;
+                VoteState::deserialize(&account.data)
+                    .ok()
+                    .map(|vote_state| (*staked, vote_state.node_pubkey))
+                    .unwrap_or((0, Pubkey::default()))
+            })
+            .collect::<HashMap<u64, Pubkey>>()
+            .drain()
+            .for_each(|(staked, node_pubkey)| {
+                node_stake_hashmap
+                    .entry(node_pubkey)
+                    .and_modify(|stake| {
+                        *stake += staked;
+                    })
+                    .or_insert(staked);
+            });
+
+        assert!(total_staked > 0);
+
+        node_stake_hashmap.iter().for_each(|(pubkey, staked)| {
+            let rent_to_be_paid =
+                (((staked * rent_to_be_distributed) as f64) / (total_staked as f64)) as u64;
+            let mut account = self.get_account(pubkey).unwrap_or_default();
+            account.lamports += rent_to_be_paid;
+            self.store_account(pubkey, &account);
+        });
+    }
+
     fn distribute_rent(&self) {
         let total_rent_collected = self.collected_rent.load(Ordering::Relaxed);
 
-        if total_rent_collected != 0 {
-            let burned_portion =
-                (total_rent_collected * u64::from(self.rent_collector.rent.burn_percent)) / 100;
-            let _rent_to_be_distributed = total_rent_collected - burned_portion;
-            // TODO: distribute remaining rent amount to validators
-            // self.capitalization.fetch_sub(burned_portion, Ordering::Relaxed);
+        let burned_portion =
```

Comment:
give rent_collector a burn() method?

Reply:
updated. Since `Rent` object is one where all computations are, and `RentCollector` is one where all actions are. Burn calculation should be in `Rent`. I have updated accordingly.